lib/redis/store/ttl.rb

```diff
@@ -11,20 +11,31 @@ def set(key, value, options = nil)
 
       def setnx(key, value, options = nil)
         if ttl = expires_in(options)
-          setnx_with_expire(key, value, ttl.to_i)
+          if options[:avoid_multi_commands]
```

Comment:
I think this decision belongs in `setnx_with_expire`, and we should create another method that takes a block which chooses between `multi` and `pipelined`, like this:

```ruby
def with_multi_or_pipelined(&block)
  return pipelined(&block) if options[:avoid_multi_commands]
  multi(&block)
end
```

Then you can leave this method untouched, and not have quite as much duplication in the private methods.

Reply:
I'll still need to pass the options hash into this method, but it's still much better. Great feedback :)